src/Umbraco.Web.UI.Client/src/common/directives/components/imaging/umbimagecrop.directive.js

```diff
@@ -275,41 +351,67 @@ angular.module("umbraco.directives")
                         }
+                    }
+
+
+                    function onViewportSizeChanged() {
+                        scope.dimensions.viewport.width = $viewport.width();
+                        scope.dimensions.viewport.height = $viewport.height();
 
-                        //sets constaints for the cropper
+                        setDimensions();
+                        setCrop();
                         setConstraints();
-                        scope.loaded = true;
-                    };
+                    }
 
 
                     // Watchers
-                    scope.$watchCollection('[width, height]', function (newValues, oldValues) {
+                    unsubscribe.push(scope.$watchCollection('[width, height, alias, forceUpdate]', function (newValues, oldValues) {
                         // We have to reinit the whole thing if
                         // one of the external params changes
                         if (newValues !== oldValues) {
-                            setDimensions($image);
+                            runtimeCrop = Utilities.copy(scope.crop);
+                            setDimensions();
+                            setCrop();
                             setConstraints();
                         }
-                    });
+                    }));
 
-                    var throttledResizing = _.throttle(function () {
+                    var throttledScale = _.throttle(() => scope.$evalAsync(() => {
                         resizeImageToScale(scope.dimensions.scale.current);
                         calculateCropBox();
-                    }, 15);
+                        saveCropBox();
+                    }), 16);
 
                     // Happens when we change the scale
-                    scope.$watch("dimensions.scale.current", function (newValue, oldValue) {
+                    unsubscribe.push(scope.$watch("dimensions.scale.current", function (newValue, oldValue) {
                         if (scope.loaded) {
-                            throttledResizing();
+                            throttledScale();
                         }
-                    });
+                    }));
+
 
                     // Init
+
+                    //if we have a max-size we will use it, to keep this backwards compatible.
+                    // I dont see this max size begin usefull, as we should aim for responsive UI.
+                    if (scope.maxSize) {
+                        element.css("max-width", parseInt(scope.maxSize, 10) + "px");
+                        element.css("max-height", parseInt(scope.maxSize, 10) + "px");
+                    }
+
                     $image.on("load", function () {
                         $timeout(function () {
-                            init($image);
+                            init();
                         });
                     });
+
+                    $window.addEventListener("resize", onViewportSizeChanged)
```

Comment:
We could use the windowResizeListener service to prevent multiple resize listeners

Reply:
Does that work better in terms of the angularJS digests? I guess my question is what this service provides that the native does not?

Reply:
I have changed so it uses the windowResizeListener :-)